### PR TITLE
#73562 - ensure Merge uses EventType and SubscriberName for matching

### PR DESCRIPTION
**Pull request by mszymeshop**
### What
This PR enables MERGE and RELOAD which were both broken for config changes. *Please ensure to question every line, this PR is ⚠️ crucial*

### Testing
Locally, the following has been tested:
- config in KV
- config in KV and Cosmos independent
- config in KV and Cosmos and KV changed and reloaded
- config in KV and Cosmos and Cosmos changed and reloaded
- config in KV and Cosmos override
- config in KV and Cosmos override changed and reloaded

I am going to perform more tests while this PR is being reviewed.

### CODE REVIEW

File: src/CaptainHook.DirectorService/Infrastructure/ReaderServicesManager.cs [hunk 1]
```diff
@@ -108,11 +108,10 @@ private async Task DeleteReaderServicesAsync(IEnumerable<string> oldNames, Cance
             }
         }
 
-        private static byte[] BuildInitializationData(SubscriberConfiguration subscriber, IEnumerable<WebhookConfig> webhooks)
+        private static byte[] BuildInitializationData(SubscriberConfiguration subscriber, IEnumerable<WebhookConfig> _)
```
Comment (dawidkeshop) [inline src/CaptainHook.DirectorService/Infrastructure/ReaderServicesManager.cs:111]: Can we just remove the second parameter it's not used?

File: src/CaptainHook.Storage.Cosmos/Models/EndpointDocument.cs [hunk 2]
```diff
@@ -81,7 +81,7 @@ internal class EndpointDocument
         /// <summary>
         /// Identifies the subscriber by event and name.
         /// </summary>
-        [JsonProperty("subscriberId")]
-        public string SubscriberId => $"{EventName}_{SubscriberName}";
+        [JsonIgnore]
+        public string SubscriberId => $"{EventName}-{SubscriberName}";
```
Comment (dawidkeshop) [inline src/CaptainHook.Storage.Cosmos/Models/EndpointDocument.cs:85]: Do we need Id to be stored? Wouldn't be enough to create it on Entity level.

File: src/CaptainHook.DirectorService/Infrastructure/ReaderServicesManager.cs (hunk 1 above)
Comment (mszymeshop) [reply src/CaptainHook.DirectorService/Infrastructure/ReaderServicesManager.cs:111]: Agreed, but I focused on making sure this PR is as readable as possible without the introduction of additional changes. I believe we already have a tech debt story to address the webconfig aspect.

File: src/CaptainHook.Storage.Cosmos/Models/EndpointDocument.cs (hunk 2 above)
Comment (mszymeshop) [reply src/CaptainHook.Storage.Cosmos/Models/EndpointDocument.cs:85]: It is not going to be stored as it is a computed property with JsonIgnore. Having said that, I agree that it shouldn't be here really as it is not part of the Document. We need to reiterate how we read the data though as at the moment the grouping is done by SubscriberId and, again, it made the most sense to include it here for the time being.

File: src/CaptainHook.DirectorService/Infrastructure/ReaderServicesManager.cs (hunk 1 above)
Comment (dawidkeshop) [reply src/CaptainHook.DirectorService/Infrastructure/ReaderServicesManager.cs:111]: OK, makes sense

File: src/CaptainHook.Storage.Cosmos/Models/EndpointDocument.cs (hunk 2 above)
Comment (dawidkeshop) [reply src/CaptainHook.Storage.Cosmos/Models/EndpointDocument.cs:85]: OK, makes sense.

**Comment by mszymeshop** [comment]
@dawidkeshop I have updated the PR so that `SubscriberId` is no longer required

**Pull request merged**